### PR TITLE
feat(list): add newIndex and oldIndex event details to calciteListOrderChange event

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #7875

## Summary

- Add more useful event details to `calciteListOrderChange` event
- Adds `oldIndex` and `newIndex` properties to the event detail
- Adds test

### CODE REVIEW

**Review by benelan — COMMENTED** [review]
What do you think about `previousIndex` and `currentIndex`? It would line up with the native event's `currentTarget`. Unless we have some precedent for new/old versus current/previous.

**Comment by driskull** [comment]
I prefer old/new vs previous/current because `current` can be a bit ambiguous.

**Comment by driskull** [comment]
Event has `originalTarget` but that property is non-standard. `previous` would go better with `next`.

File: packages/calcite-components/src/components/list/list.e2e.ts [hunk 1]
```diff
@@ -616,24 +631,30 @@ describe("calcite-list", () => {
 
       let totalMoves = 0;
 
-      const eventSpy = await page.spyOnEvent("calciteListOrderChange");
+      // Workaround for page.spyOnEvent() failing due to drag event payload being serialized and there being circular JSON structures from the payload elements. See: https://github.com/Esri/calcite-design-system/issues/7643
```
Comment (jcfranco) [inline packages/calcite-components/src/components/list/list.e2e.ts:634]: Nice!

File: packages/calcite-components/src/components/list/list.e2e.ts [hunk 2]
```diff
@@ -482,8 +485,12 @@ describe("calcite-list", () => {
       // Workaround for page.spyOnEvent() failing due to drag event payload being serialized and there being circular JSON structures from the payload elements. See: https://github.com/Esri/calcite-design-system/issues/7643
       await page.$eval("calcite-list", (list: HTMLCalciteListElement) => {
         (window as TestWindow).calledTimes = 0;
```
Comment (jcfranco) [inline packages/calcite-components/src/components/list/list.e2e.ts:487]: Suggestion: can you store the typed window in a local `win` variable or similar to reduce typecasting lines?

File: packages/calcite-components/src/utils/sortableComponent.ts [hunk 3]
```diff
@@ -5,6 +5,8 @@ export interface DragDetail {
   toEl: HTMLElement;
```
Comment (jcfranco) [inline packages/calcite-components/src/utils/sortableComponent.ts:5]: Sidebar: we should probably add a conventions entry about meaningful event details and when they should be allowed. This is a perfect example of how this info helps add context behind an event that would be hard to represent in another way.

**Review by jcfranco — APPROVED** [review]
Awesome! ✨👶📇👴📇 ✨

File: packages/calcite-components/src/utils/sortableComponent.ts (hunk 3 above)
Comment (driskull) [reply packages/calcite-components/src/utils/sortableComponent.ts:5]: Yes a convention document on this would be super useful!

**Pull request merged**